### PR TITLE
feat(events): commande de refresh dédiée (upsert + purge) + planification hebdo

### PR DESCRIPTION
**Pull request by vincentchalamon**
Closes #985.

> ⚠️ **Sommet du stack #987 (`feature/975`) → #996 (`feature/984`) → #996-985.** À merger **en dernier**, après #987 puis #996. Cascade `--onto` via `gh stack rebase` au fil des squash-merges.

## Résumé
Cycle de vie temporel des événements : upsert + purge, isolé du modèle append-only (ADR-049).

- **`EventsPromotion`** (nouveau) : SQL de promotion events — `INSERT … ON CONFLICT (id) DO UPDATE` des champs mutables (name, category, start/end_date, url, description, price_min, source, tags, geom, last_seen_at ; `zone` exclu = provenance), clippé à la zone, suivi de `DELETE … WHERE end_date < :today` dans la **même transaction**. `:today` validé `YYYY-MM-DD` et injecté en littéral (jamais `now()`/`CURRENT_DATE`). Compteurs insert/update via `xmax`.
- **`EventsRefreshCommand`** + `provisioner/bin/events-refresh` (nouveaux) : commande `events-refresh` autonome, découvre les zones ouvertes via `osm.zones`, download une fois par flux, upsert+purge par zone, continue-on-error, partage le lock de provisioning. Options `--zone`, `--dry-run`. Date de purge pinnée en `Europe/Paris`.
- **`EventsRefreshSourceInterface`** (nouveau) implémenté par les 2 importers ; events retirés du `ZonePromotion` append-only (places uniquement). `provision <zone>` délègue au même chemin.
- **Makefile** : cible `events-refresh` (override entrypoint, pattern `provision-override`).

## Docs
ADR-051 §4 étendu (upsert+purge/scheduling, pourquoi pas d'osm-cron) ; nouveau `docs/runbooks/events-refresh.md` ; `deployment.md`, `zone-opening.md`, README runbooks.

## Vérification (legs QA)
Provisioner PHPUnit `234 tests, 1 skipped` (dont `EventsPromotionExecutionTest` DB-backed : moved-date-updates-not-duplicates, purge incl. event passé juste upserté, clip in-zone, idempotence + report ; `EventsRefreshCommandTest` 7 tests). **Négatif prouvé** : purge `WHERE false` → rouge ; upsert `DO NOTHING` → rouge ; restaurés. PHPStan L9 `No errors` ; Rector `[OK]` (1 autofix) ; PHP-CS-Fixer `0 of 43` ; markdownlint `0` ; liens MD internes `0` ; `api/tests/Unit 1412`.

## Auto-critique
Aucun DTO/dep. **Tâche planifiée Coolify = étape manuelle prod** (cmd `php -d memory_limit=512M bin/events-refresh`, cron `0 3 * * 0`), documentée, non automatisée. Recette réelle events-refresh sur zone = post-merge (flux live + zones ouvertes). L'invocation `run --rm provisioner events-refresh` de l'issue ne marche pas telle quelle (`bin/provision` en single-command) → forme documentée override l'entrypoint. Playwright N/A (provisioner-only).

## Claude Review

**Summary:** Correctness, security, and architecture remain solid: the upsert+purge SQL is properly parameterized (`:today` regex-validated as `YYYY-MM-DD`, zone/source values escaped via `ZonePromotion::literal()`, `--zone` CLI input checked against the live open-zones registry before it reaches SQL), the shared provisioning lock prevents concurrent runs, and behavior is covered by a DB-backed execution test (`EventsPromotionExecutionTest`) plus a command-orchestration test (`EventsRefreshCommandTest`). The second commit on this PR fixed the previously-flagged full-flux-extraction issue (`extract()` now takes `onlyTables` and the events refresh no longer spills the ~390k-object place dataset to disk), with a guard test added. One previously-flagged performance thread remains open (see below).

**Resolved threads:** 1. The `DataTourismeImporter.php` full-4-table-`extract()` thread is fixed by this PR's second commit (`extract($zipPath, $workDir, ['events'])` + `eventsRefreshExtractsEventsOnlyAndNeverSpillsThePlaceTables` guard test) and has been resolved.

### Review checklist
- [x] Code respects the project architecture (ADR-051 §4 upsert+purge kept isolated from the append-only ADR-049 model; `EventsRefreshSourceInterface` cleanly separates the two importers from the command)
- [x] SOLID principles and Law of Demeter followed
- [x] Design patterns used where appropriate
- [x] Tests cover new/changed cases (DB-backed execution test + command orchestration test, including negative/mutation-style checks per the PR description)
- [x] Documentation is up to date (ADR-051, runbook, deployment.md, zone-opening.md all updated consistently)
- [ ] Dependent tickets accounted for — this PR is the tail of a 3-PR stack (#987 → #996 → this one); merge order matters, not a code issue

### Inline comments
No new inline comments. 1 previously-open thread still applies, left open:
1. `provisioner/src/EventsRefreshCommand.php:177` — the *global* events purge (`DELETE FROM tourism.events WHERE end_date < :today`) re-runs once per open zone per source (2×N per refresh run) instead of once per source per invocation. Idempotent and index-assisted so not wrong, just avoidable full-table-scan work that scales with the number of open zones.

**Commit reviewed:** `110bb4f8fb8f5c50c7981ef9501b3e58134bc472`

### CODE REVIEW

File: provisioner/src/DataTourismeImporter.php [hunk 1]
```diff
@@ -268,12 +312,57 @@ private function zoneHasGeometry(string $workDir, string $zoneSlug): bool
...
+     * schema. The places are parsed but not loaded: the standalone refresh writes
+     * `tourism.events` alone, never the append-only place tables (ADR-051 §4).
+     *
+     * @throws ImportFailedException
+     */
+    public function stageEventsForRefresh(string $workDir): string
+    {
+        $zipPath = $workDir.'/datatourisme-flux.zip';
+        $this->download($zipPath);
+        $copyFiles = $this->extract($zipPath, $workDir);
```
[the remark this answers is not in the document]
Comment (vincentchalamon) [reply provisioner/src/DataTourismeImporter.php]: Corrigé (110bb4f8) : `extract()` accepte désormais `?array $onlyTables` ; `stageEventsForRefresh()` appelle `extract($zip, $workDir, ['events'])`, donc les trois fichiers COPY de lieux ne sont plus écrits sur disque lors du refresh hebdo. Test de garde ajouté (`eventsRefreshExtractsEventsOnlyAndNeverSpillsThePlaceTables`).

File: provisioner/src/EventsRefreshCommand.php [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+        } catch (ImportFailedException $importFailedException) {
+            $this->fail($io, \sprintf('%s feed download/parse failed: %s', $source->label(), $importFailedException->getMessage()));
+
+            return Command::FAILURE;
+        }
+
+        $outcome = Command::SUCCESS;
+        foreach ($zones as $zone) {
+            try {
+                $source->promoteEventsForZone($staging, $zone, $today);
```
[the remark this answers is not in the document]
Comment (vincentchalamon) [reply provisioner/src/EventsRefreshCommand.php:177]: Non actionné volontairement. La purge dans la même transaction que l'upsert de chaque zone est un choix délibéré d'ADR-051 §4 : un événement déjà terminé est purgé atomiquement, y compris s'il vient d'être upserté. La sortir de la boucle (une fois par source) affaiblirait cette propriété transactionnelle pour un gain marginal : le `DELETE ... end_date < today` porte sur `tourism.events` (couche modeste, purgée en continu), pas sur les ~390k lieux — le seq-scan répété est négligeable et idempotent, comme noté. Le vrai coût (écriture des lieux) était le thread ci-dessus, lui corrigé. Ré-ouvrable en optimisation dédiée si le volume d'events par zone le justifie un jour.
